spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/actions/SparkBinPackPositionDeletesRewriter.java

```diff
@@ -138,4 +142,30 @@ private Dataset<Row> dataFiles(Types.StructType partitionType, StructLike partit
       return SparkTableUtil.loadMetadataTable(spark, table(), MetadataTableType.DATA_FILES);
     }
   }
+
+  /**
+   * If necessary, convert the partition data (from avro) to its logical type, to use in the spark
+   * queries. See https://iceberg.apache.org/spec/#avro.
+   *
+   * @param value partition object
+   * @param typeId iceberg type
+   * @return object converted to its logical type
+   */
+  private Object convertPartitionValue(Object value, Type.TypeID typeId) {
+    switch (typeId) {
```

Comment:
The expected java class is from: https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/types/Type.java#L31

Reply:
I understand there is a type mismatch but don't we have to convert internal Iceberg values for a given type to a Spark value? Why Avro?

Reply:
We are getting back Iceberg literals we need to come up with Spark literals. We have `PartitionData` under the hood that is populated by Avro but that should not matter at all. We need to build an reverser conversion similar to what we already have in `SparkValueConverter`.

Reply:
Yea, makes sense, its not really to do with Avro, but indeed we need an Iceberg => Spark direction, just as we have a Spark => Iceberg direction in SparkValueConverter.  Added that in latest commit.